pkg/verification/service/verification_service.go

```diff
@@ -0,0 +1,274 @@
+// the user will receive a verification SMS.  The UserSignup resource is updated with a number of annotations in order
+// to manage the phone verification process and protect against system abuse.
+func (s *ServiceImpl) InitVerification(ctx *gin.Context, userID string, e164PhoneNumber string) error {
+	var initError error
+
+	signup, err := s.Services().SignupService().GetUserSignup(userID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Error(ctx, err, "usersignup not found")
+			return errors.NewNotFoundError(err, "usersignup not found")
+		}
+		log.Error(ctx, err, "error retrieving usersignup")
+		return errors.NewInternalError(err, fmt.Sprintf("error retrieving usersignup: %s", userID))
+	}
+
+	// check that verification is required before proceeding
+	if signup.Spec.VerificationRequired == false {
+		log.Info(ctx, fmt.Sprintf("phone verification attempted for user without verification requirement: %s", userID))
+		return errors.NewBadRequest("forbidden request", "verification code will not be sent")
+	}
+
+	// Check if the provided phone number is already being used by another user
+	err = s.Services().SignupService().PhoneNumberAlreadyInUse(userID, e164PhoneNumber)
+	if err != nil {
+		if apierrors.IsForbidden(err) {
+			log.Errorf(ctx, err, "phone number already in use, cannot register using phone number: %s", e164PhoneNumber)
+			return errors.NewForbiddenError("phone number already in use", fmt.Sprintf("cannot register using phone number: %s", e164PhoneNumber))
+		}
+		log.Error(ctx, err, "error while looking up users by phone number")
+		return errors.NewInternalError(err, "could not lookup users by phone number")
+	}
+
+	// calculate the phone number hash
+	md5hash := md5.New()
+	// Ignore the error, as this implementation cannot return one
+	_, _ = md5hash.Write([]byte(e164PhoneNumber))
+	phoneHash := hex.EncodeToString(md5hash.Sum(nil))
+
+	if signup.Labels == nil {
+		signup.Labels = map[string]string{}
+	}
+	signup.Labels[v1alpha1.UserSignupUserPhoneHashLabelKey] = phoneHash
+
+	// read the current time
+	now := time.Now()
+
+	// If 24 hours has passed since the verification timestamp, then reset the timestamp and verification attempts
+	ts, parseErr := time.Parse(TimestampLayout, signup.Annotations[v1alpha1.UserSignupVerificationInitTimestampAnnotationKey])
+	if parseErr != nil || (parseErr == nil && now.After(ts.Add(24*time.Hour))) {
+		// Set a new timestamp
+		signup.Annotations[v1alpha1.UserSignupVerificationInitTimestampAnnotationKey] = now.Format(TimestampLayout)
+		signup.Annotations[v1alpha1.UserSignupVerificationCounterAnnotationKey] = "0"
+	}
+
+	// get the verification counter (i.e. the number of times the user has initiated phone verification within
+	// the last 24 hours)
+	verificationCounter := signup.Annotations[v1alpha1.UserSignupVerificationCounterAnnotationKey]
+	var counter int
+	dailyLimit := s.config.GetVerificationDailyLimit()
+	if verificationCounter == "" {
+		counter = 0
```

Comment:
can't we set counter to 0 and remove some of this nesting? 
```
counter := 0
if verifiationCounter != "" {
   counter, err = strconv.Atoi(verificationCounter)
   ...
}
```

Reply:
Isn't this your code? ;)  All I did here was change the variable name.

Reply:
does it matter whose code it is :-) i think it can be done better and if we're re-factoring may as well do so now, right?

Reply:
My point here isn't about whose code this is, it's that this is not new code that has been added for the purposes of this PR.  Making this change here is out of scope of the PR however I've gone ahead and simplified it by removing the first condition.